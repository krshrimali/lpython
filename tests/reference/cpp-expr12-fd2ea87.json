{
    "basename": "cpp-expr12-fd2ea87",
    "cmd": "lpython --no-color --show-cpp {infile}",
    "infile": "tests/expr12.py",
    "infile_hash": "bad9ac6e0956fddb636f4e806f4a97c27396adc2416640bd291d2dc8",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "cpp-expr12-fd2ea87.stdout",
    "stdout_hash": "0ec78f2ed6092a1cfb65a4f59a7ef92994bf878daf52d11a5d2b27a5",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}