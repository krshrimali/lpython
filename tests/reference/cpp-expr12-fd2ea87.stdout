#include <iostream>
#include <string>
#include <vector>
#include <cassert>
#include <cmath>
#include <complex>
#include <Kokkos_Core.hpp>
#include <lfortran_intrinsics.h>

template <typename T>
Kokkos::View<T*> from_std_vector(const std::vector<T> &v)
{
    Kokkos::View<T*> r("r", v.size());
    for (size_t i=0; i < v.size(); i++) {
        r(i) = v[i];
    }
    return r;
}


struct dimension_descriptor
{
    int32_t lower_bound, length;
};
// Forward declarations
void _lpython_main_program();
int32_t check();
void main0();
int32_t test(int a, int b);
namespace {
}

// Implementations
void _lpython_main_program()
{
    main0();
}

int32_t check()
{
    int _lpython_return_variable;
    int a;
    a = test(2, 2);
    _lpython_return_variable = a;
    return _lpython_return_variable;
}

void main0()
{
    int x;
    x = check();
}

int32_t test(int a, int b)
{
    int _lpython_return_variable;
    _lpython_return_variable = std::pow(a, b);
    return _lpython_return_variable;
}

namespace {

void main2() {
    _lpython_main_program();
}

}
int main(int argc, char* argv[])
{
    Kokkos::initialize(argc, argv);
    main2();
    Kokkos::finalize();
    return 0;
}
